Source/Plugins/EditorBase/PropertyEditors/AddComponentPropertyEditor.cs

```diff
@@ -0,0 +1,113 @@
+﻿using System.Drawing;
+using System.Linq;
+using System.Windows.Forms;
+using AdamsLair.WinForms.PropertyEditing;
+using Duality.Editor.Forms;
+using Duality.Editor.UndoRedoActions;
+using ButtonState = AdamsLair.WinForms.Drawing.ButtonState;
+
+namespace Duality.Editor.Plugins.Base.PropertyEditors
+{
+	/// <summary>
+	/// A <see cref="PropertyEditor"/> that provides a button
+	/// for adding a component to a <see cref="GameObject"/>
+	/// </summary>
+	public class AddComponentPropertyEditor : PropertyEditor
```

Comment:
Works as expected, but for improved usability, the editor should also react to the return key in order to "press" the button - that way the Object Inspector remains usable completely via keyboard. There should be a virtual method you can override to react to keyboard input while the editor is selected.